Backend/Routes/users.js

```diff
@@ -0,0 +1,85 @@
+import express from 'express';
+import bcrypt from 'bcrypt';
+import { User } from '../models/user.js';
+import { Op } from 'sequelize';
+import { PrismaClient } from '@prisma/client';
+
+const router = express.Router();
+
+// Route for user registration
+router.post('/users/signup', async (req, res) => {
+
+  const { name, password, email, userRole } = req.body;
+
+
+  try {
+    // Check if email already exists
+    const existingUser = await User.findOne({
+      where: {
+        [Op.or]: [{ email }]
+      }
+    });
+
+    if (existingUser) {
+      return res.status(400).json({ error: 'Email already exists' });
+    }
+
+    // Encrypt the password
+    const hashedPassword = await bcrypt.hash(password, 10);
+
+    // Create a new user
+    const newUser = await User.create({
+      name,
+      email,
+      password: hashedPassword,
+      userRole
+    });
+
+    // Set the user in the session
+    req.session.user = newUser;
+
+    // Return the user data in the response
+    res.json({ user: newUser });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: 'Server error' });
+  }
+});
+
+
+// Route for user login
+router.post('/users/login', async (req, res) => {
+  const { email, password } = req.body;
+
+  try {
+    // Find the user by email
+    const user = await User.findOne({ where: { email } });
+
+    if (!user) {
+      return res.status(401).json({ error: 'Invalid email or password' });
```

Comment:
error: 'Invalid email' ? and ln66 for error: 'Invalid password' ?

Reply:
Resolved. I have updated it, one for email, one for password.